### ISSUE TITLE
Installation issue: q-e-sirius +openmp fails with ^fftw@3

### ISSUE THREAD

**Issue by lorisercole** [LLNL/spack#39726]
### Steps to reproduce the issue

```console
# spack spec -I q-e-sirius@develop-ristretto+openmp ^fftw@3
Input spec
--------------------------------
 -   q-e-sirius@develop-ristretto+openmp
 -       ^fftw@3

Concretized
--------------------------------
 -   q-e-sirius@develop-ristretto%gcc@12.2.0~elpa~hdf5~ipo~libxc+mpi+openmp~ptrace~scalapack~sirius_apps~trace build_system=cmake build_type=Release generator=make patches=3dec741 arch=linux-almalinux8-x86_64_v3
[+]      ^cmake@3.26.3%gcc@12.2.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-almalinux8-x86_64_v3
[+]          ^curl@8.1.2%gcc@12.2.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-almalinux8-x86_64_v3
[+]              ^nghttp2@1.52.0%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^ncurses@6.4%gcc@12.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^zlib@1.2.13%gcc@12.2.0+optimize+pic+shared build_system=makefile arch=linux-almalinux8-x86_64_v3
[+]      ^git@2.41.0%gcc@12.2.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^autoconf@2.69%gcc@12.2.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-almalinux8-x86_64_v3
[+]          ^automake@1.16.5%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^expat@2.5.0%gcc@12.2.0+libbsd build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^libbsd@0.11.7%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]                  ^libmd@1.0.4%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^gettext@0.21.1%gcc@12.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^bzip2@1.0.8%gcc@12.2.0~debug~pic+shared build_system=generic arch=linux-almalinux8-x86_64_v3
[+]              ^libxml2@2.10.3%gcc@12.2.0~python build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^tar@1.34%gcc@12.2.0 build_system=autotools zip=pigz arch=linux-almalinux8-x86_64_v3
[+]                  ^pigz@2.7%gcc@12.2.0 build_system=makefile arch=linux-almalinux8-x86_64_v3
[+]                  ^zstd@1.5.5%gcc@12.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-almalinux8-x86_64_v3
[+]              ^xz@5.4.1%gcc@12.2.0~pic build_system=autotools libs=shared,static arch=linux-almalinux8-x86_64_v3
[+]          ^libiconv@1.17%gcc@12.2.0 build_system=autotools libs=shared,static arch=linux-almalinux8-x86_64_v3
[+]          ^libidn2@2.3.4%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^libunistring@1.1%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^libtool@2.4.7%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^m4@1.4.19%gcc@12.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-almalinux8-x86_64_v3
[+]              ^diffutils@3.9%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^libsigsegv@2.14%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^openssh@9.3p1%gcc@12.2.0+gssapi build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^krb5@1.20.1%gcc@12.2.0+shared build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]                  ^bison@3.8.2%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]                  ^findutils@4.9.0%gcc@12.2.0 build_system=autotools patches=440b954 arch=linux-almalinux8-x86_64_v3
[+]              ^libedit@3.1-20210216%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]              ^libxcrypt@4.4.35%gcc@12.2.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-almalinux8-x86_64_v3
[+]          ^openssl@3.1.2%gcc@12.2.0~docs+shared build_system=generic certs=mozilla arch=linux-almalinux8-x86_64_v3
[+]              ^ca-certificates-mozilla@2023-05-30%gcc@12.2.0 build_system=generic arch=linux-almalinux8-x86_64_v3
[+]          ^pcre2@10.42%gcc@12.2.0~jit+multibyte build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^perl@5.38.0%gcc@12.2.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-almalinux8-x86_64_v3
[+]              ^berkeley-db@18.1.40%gcc@12.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-almalinux8-x86_64_v3
[+]              ^gdbm@1.23%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]                  ^readline@8.2%gcc@12.2.0 build_system=autotools patches=bbf97f1 arch=linux-almalinux8-x86_64_v3
[+]      ^gmake@4.4.1%gcc@12.2.0~guile build_system=autotools arch=linux-almalinux8-x86_64_v3
[e]      ^hpcx-mpi@2.15%gcc@12.2.0 build_system=generic arch=linux-almalinux8-x86_64_v3
[+]      ^pkgconf@1.9.5%gcc@12.2.0 build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]      ^sirius@develop%gcc@12.2.0+apps~boost_filesystem~cuda~elpa+fortran~ipo~magma+memory_pool~nlcglib+openmp+profiler~python~rocm~scalapack+shared~single_precision~tests~vdwxc~wannier90 build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
[+]          ^costa@2.2.2%gcc@12.2.0~apps~benchmarks~ipo~profiling~scalapack+shared~tests build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
[+]          ^fftw@3.3.10%gcc@12.2.0+mpi~openmp~pfft_patches build_system=autotools precision=double,float arch=linux-almalinux8-x86_64_v3
[+]          ^gsl@2.7.1%gcc@12.2.0~external-cblas build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^hdf5@1.14.2%gcc@12.2.0~cxx~fortran+hl~ipo~java~map+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
[+]          ^libxc@6.1.0%gcc@12.2.0~cuda~kxc~lxc+shared build_system=autotools arch=linux-almalinux8-x86_64_v3
[+]          ^openblas@0.3.23%gcc@12.2.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-almalinux8-x86_64_v3
[+]          ^spfft@1.0.6%gcc@12.2.0~cuda~fortran~gpu_direct~ipo+mpi+openmp~rocm~single_precision~static build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
[+]          ^spglib@2.0.2%gcc@12.2.0~ipo+openmp build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
[+]          ^spla@1.5.5%gcc@12.2.0~cuda~fortran~ipo+openmp~rocm~static build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
[+]          ^umpire@2022.03.1%gcc@12.2.0+c~cuda+device_alloc~deviceconst+examples~fortran~ipo~numa~openmp~rocm+shared build_system=cmake build_type=Release generator=make tests=none arch=linux-almalinux8-x86_64_v3
[+]              ^blt@0.5.3%gcc@12.2.0 build_system=generic arch=linux-almalinux8-x86_64_v3
[+]              ^camp@2022.10.1%gcc@12.2.0~cuda~ipo~openmp~rocm~tests build_system=cmake build_type=Release generator=make arch=linux-almalinux8-x86_64_v3
```

### Error message

The problems seems to be related to using `fftw@3` without openmp support.
@simonpintarelli 

<details><summary>Error message</summary>

<pre>
[+] /anfhome/spack/opt/spack/[padded-to-128-chars]/linux-almalinux8-x86_64_v3/gcc-12.2.0/sirius-develop-tah7cmi2rf4nrruo4lck57iggtklhf7n
==> Installing q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj [46/46]
==> No binary for q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj found: installing from source
==> Using cached archive: /anfhome/spack/var/spack/cache/_source-cache/archive/3d/3dec7410f0f6706765870da6d2bcf40ed9a847f2c5a789aef71e71003f1eca59
==> Applied patch https://raw.githubusercontent.com/matt-chan/q-e-sirius/offline/offline.patch
==> q-e-sirius: Executing phase: 'cmake'
==> q-e-sirius: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/gmake-4.4.1-hex2jbxiqacg7gmcutq6wkniebukvlgi/bin/make' '-j16'

23 errors found in build log:
     1586    /anfhome/spack/lib/spack/env/gcc/gfortran -Wl,-rpath -Wl,/opt/hpcx-v2.15-gcc-MLNX_OFED_LINUX-5-redhat8-cuda12-gdrcopy2-nccl2.17-x86_64/ompi/lib -Wl,--enable-new-dtags -pthread  -fallow-argument-mismatch -O3 -DNDEBUG -O3 CMakeFiles/test_qe_fftx_fft_scatter_mod_gpu.dir/test_fft_scatter_mod_gpu.f90.o -o ../bin/test_qe_fftx_fft_scatter
             _mod_gpu.x  -Wl,-rpath,/opt/hpcx-v2.15-gcc-MLNX_OFED_LINUX-5-redhat8-cuda12-gdrcopy2-nccl2.17-x86_64/ompi/lib:/anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/fftw-3.3.10-ydck7tigeyopn5vsn6v72hpfdqj3djmw/lib ../../l
             ib/libqe_fftx_test_common.a ../../lib/libqe_fftx.a /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-haswell/gcc-8.5.0/gcc-12.2.0-tfhxftxrkhhrmhq4l4kp3sdjkm7kjttt/lib64/libgomp.so /lib64/libpthread.so /opt/hpcx-v2.15-gcc-MLNX_OFED_LINUX-
             5-redhat8-cuda12-gdrcopy2-nccl2.17-x86_64/ompi/lib/libmpi_usempif08.so /opt/hpcx-v2.15-gcc-MLNX_OFED_LINUX-5-redhat8-cuda12-gdrcopy2-nccl2.17-x86_64/ompi/lib/libmpi_usempi_ignore_tkr.so /opt/hpcx-v2.15-gcc-MLNX_OFED_LINUX-5-redhat8-cuda12-gdrcopy2-nccl2.17-x86_64/ompi/lib/libmpi_mpifh.so /opt/hpcx-v2.15-gcc-MLNX_OFED_LINUX-5-redhat
             8-cuda12-gdrcopy2-nccl2.17-x86_64/ompi/lib/libmpi.so /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/fftw-3.3.10-ydck7tigeyopn5vsn6v72hpfdqj3djmw/lib/libfftw3.so
     1587    make[2]: Entering directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
     1588    cd /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4 && /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ctjz3lzao
             oy6w3ammknh56xysq/bin/cmake -E cmake_depends "Unix Makefiles" /mnt/scratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-src /mnt/scratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-src/UtilXlib /mnt/nvme/tmpscratch/root/spack-st
             age/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4 /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4/UtilXlib /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbs
             ksnbkqdjpcwewszxj/spack-build-4ktzyk4/UtilXlib/CMakeFiles/test_qe_utilx_mp_bcast_i1.dir/DependInfo.cmake --color=
     1589    make[2]: Leaving directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
     1590    /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/gmake-4.4.1-hex2jbxiqacg7gmcutq6wkniebukvlgi/bin/make  -f UtilXlib/CMakeFiles/test_qe_utilx_mp_bcast_i1.dir/build.make UtilXlib/CMakeFiles/test_qe_utilx_mp_bcast_i1.d
             ir/build
     1591    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cfft3ds':
  >> 1592    fft_scalar.FFTW3.f90:(.text+0x359): undefined reference to `fftw_init_threads'
  >> 1593    fft_scalar.FFTW3.f90:(.text+0x37d): undefined reference to `fftw_plan_with_nthreads'
     1594    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cfft3d':
  >> 1595    fft_scalar.FFTW3.f90:(.text+0xdc9): undefined reference to `fftw_init_threads'
  >> 1596    fft_scalar.FFTW3.f90:(.text+0xddf): undefined reference to `fftw_plan_with_nthreads'
     1597    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cft_2xy':
  >> 1598    fft_scalar.FFTW3.f90:(.text+0x1abb): undefined reference to `fftw_init_threads'
  >> 1599    fft_scalar.FFTW3.f90:(.text+0x1adf): undefined reference to `fftw_plan_with_nthreads'
     1600    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cft_1z':
  >> 1601    fft_scalar.FFTW3.f90:(.text+0x3f99): undefined reference to `fftw_init_threads'
  >> 1602    fft_scalar.FFTW3.f90:(.text+0x3faf): undefined reference to `fftw_plan_with_nthreads'
     1603    make[2]: Entering directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
  >> 1604    collect2: error: ld returned 1 exit status
  >> 1605    make[2]: *** [FFTXlib/tests/CMakeFiles/qe_fftx_test.dir/build.make:107: FFTXlib/tests/bin/qe_fftx_test.x] Error 1
     1606    make[2]: Leaving directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
  >> 1607    make[1]: *** [CMakeFiles/Makefile2:1304: FFTXlib/tests/CMakeFiles/qe_fftx_test.dir/all] Error 2
     1608    make[1]: *** Waiting for unfinished jobs....
     1609    /mnt/scratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-src/LAXlib/cdiaghg.f90:196:18:
     1610
     1611      193 |   CALL MPI_BCAST( e, SIZE(e), MPI_DOUBLE_PRECISION, root_bgrp, intra_bgrp_comm, info )
     1612          |                  2
     1613    ......

     ...

     1789    /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ctjz3lzaooy6w3ammknh56xysq/bin/cmake -E touch external/CMakeFiles/qe_wannier90.dir/wannier90/src/ws_distance.F90.o.provides.build
     1790    [ 12%] Building Fortran object external/CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/postw90_common.F90.o
     1791    cd /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4/external && /anfhome/spack/lib/spack/env/gcc/gfortran -DHAVE_MALLINFO -DOMPI_SKIP_MPICXX -DQE_NO_CONFIG_H -D__FFTW3 -D__MPI -D__SIRIUS -I/mnt/scratch/root/spack-stage/spack-stage-q-e-sirius-develop-
             ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-src/include -I/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4/external/mod/qe_wannier90 -fallow-argument-mismatch -O3 -DNDEBUG -O3 -Jmod/qe_wannier90 -cpp -c /mnt/scratch/root/spack-stage/spack-stage-
             q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-src/external/wannier90/src/postw90/postw90_common.F90 -o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/postw90_common.F90.o
     1792    /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ctjz3lzaooy6w3ammknh56xysq/bin/cmake -E cmake_copy_f90_mod external/mbd/src/mbd_rpa.mod external/mbd/src/CMakeFiles/mbd.dir/mbd_rpa.mod.stamp GNU
     1793    /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ctjz3lzaooy6w3ammknh56xysq/bin/cmake -E touch external/mbd/src/CMakeFiles/mbd.dir/mbd_rpa.F90.o.provides.build
     1794    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cfft3ds':
  >> 1795    fft_scalar.FFTW3.f90:(.text+0x359): undefined reference to `fftw_init_threads'
  >> 1796    fft_scalar.FFTW3.f90:(.text+0x37d): undefined reference to `fftw_plan_with_nthreads'
     1797    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cfft3d':
  >> 1798    fft_scalar.FFTW3.f90:(.text+0xdc9): undefined reference to `fftw_init_threads'
  >> 1799    fft_scalar.FFTW3.f90:(.text+0xddf): undefined reference to `fftw_plan_with_nthreads'
     1800    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cft_2xy':
  >> 1801    fft_scalar.FFTW3.f90:(.text+0x1abb): undefined reference to `fftw_init_threads'
  >> 1802    fft_scalar.FFTW3.f90:(.text+0x1adf): undefined reference to `fftw_plan_with_nthreads'
     1803    ../../lib/libqe_fftx.a(fft_scalar.FFTW3.f90.o): In function `__fft_scalar_fftw3_MOD_cft_1z':
  >> 1804    fft_scalar.FFTW3.f90:(.text+0x3f99): undefined reference to `fftw_init_threads'
  >> 1805    fft_scalar.FFTW3.f90:(.text+0x3faf): undefined reference to `fftw_plan_with_nthreads'
  >> 1806    collect2: error: ld returned 1 exit status
  >> 1807    make[2]: *** [FFTXlib/tests/CMakeFiles/test_qe_fftx_fwinv_gpu.dir/build.make:108: FFTXlib/bin/test_qe_fftx_fwinv_gpu.x] Error 1
     1808    make[2]: Leaving directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
  >> 1809    make[1]: *** [CMakeFiles/Makefile2:1411: FFTXlib/tests/CMakeFiles/test_qe_fftx_fwinv_gpu.dir/all] Error 2
     1810    /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ctjz3lzaooy6w3ammknh56xysq/bin/cmake -E cmake_copy_f90_mod external/mbd/src/mbd_scs.mod external/mbd/src/CMakeFiles/mbd.dir/mbd_scs.mod.stamp GNU
     1811    /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ctjz3lzaooy6w3ammknh56xysq/bin/cmake -E touch external/mbd/src/CMakeFiles/mbd.dir/mbd_scs.f90.o.provides.build
     1812    [ 12%] Linking Fortran static library ../lib/libqe_upflib.a
     1813    cd /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4/upflib && /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ct
             jz3lzaooy6w3ammknh56xysq/bin/cmake -P CMakeFiles/qe_upflib.dir/cmake_clean_target.cmake
     1814    cd /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4/upflib && /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4ct
             jz3lzaooy6w3ammknh56xysq/bin/cmake -E cmake_link_script CMakeFiles/qe_upflib.dir/link.txt --verbose=1
     1815    /bin/ar qc ../lib/libqe_upflib.a CMakeFiles/qe_upflib.dir/atom.f90.o CMakeFiles/qe_upflib.dir/atomic_number.f90.o CMakeFiles/qe_upflib.dir/dqvan2.f90.o CMakeFiles/qe_upflib.dir/gen_us_dj.f90.o CMakeFiles/qe_upflib.dir/gen_us_dy.f90.o CMakeFiles/qe_upflib.dir/init_us_0.f90.o CMakeFiles/qe_upflib.dir/init_us_b0.f90.o CMakeFiles/qe_up
             flib.dir/init_us_1.f90.o CMakeFiles/qe_upflib.dir/init_tab_atwfc.f90.o CMakeFiles/qe_upflib.dir/init_tab_beta.f90.o CMakeFiles/qe_upflib.dir/init_tab_qrad.f90.o CMakeFiles/qe_upflib.dir/interp_atwfc.f90.o CMakeFiles/qe_upflib.dir/init_us_2_base.f90.o CMakeFiles/qe_upflib.dir/gen_us_dj_gpu.f90.o CMakeFiles/qe_upflib.dir/gen_us_dy_gp
             u.f90.o CMakeFiles/qe_upflib.dir/gth.f90.o CMakeFiles/qe_upflib.dir/paw_variables.f90.o CMakeFiles/qe_upflib.dir/pseudo_types.f90.o CMakeFiles/qe_upflib.dir/qvan2.f90.o CMakeFiles/qe_upflib.dir/radial_grids.f90.o CMakeFiles/qe_upflib.dir/read_cpmd.f90.o CMakeFiles/qe_upflib.dir/read_fhi.f90.o CMakeFiles/qe_upflib.dir/read_ncpp.f90.
             o CMakeFiles/qe_upflib.dir/read_ps.f90.o CMakeFiles/qe_upflib.dir/read_upf_new.f90.o CMakeFiles/qe_upflib.dir/read_upf_v1.f90.o CMakeFiles/qe_upflib.dir/read_uspp.f90.o CMakeFiles/qe_upflib.dir/spinor.f90.o CMakeFiles/qe_upflib.dir/sph_ind.f90.o CMakeFiles/qe_upflib.dir/sph_bes.f90.o CMakeFiles/qe_upflib.dir/splinelib.f90.o CMakeFi
             les/qe_upflib.dir/simpsn.f90.o CMakeFiles/qe_upflib.dir/upf_auxtools.f90.o CMakeFiles/qe_upflib.dir/upf_const.f90.o CMakeFiles/qe_upflib.dir/upf_error.f90.o CMakeFiles/qe_upflib.dir/upf_invmat.f90.o CMakeFiles/qe_upflib.dir/upf_io.f90.o CMakeFiles/qe_upflib.dir/upf_ions.f90.o CMakeFiles/qe_upflib.dir/upf_kinds.f90.o CMakeFiles/qe_u
             pflib.dir/upf_spinorb.f90.o CMakeFiles/qe_upflib.dir/upf_params.f90.o CMakeFiles/qe_upflib.dir/upf_parallel_include.f90.o CMakeFiles/qe_upflib.dir/upf_to_internal.f90.o CMakeFiles/qe_upflib.dir/upf_utils.f90.o CMakeFiles/qe_upflib.dir/uspp.f90.o CMakeFiles/qe_upflib.dir/uspp_data.f90.o CMakeFiles/qe_upflib.dir/uspp_param.f90.o CMak
             eFiles/qe_upflib.dir/write_upf_new.f90.o CMakeFiles/qe_upflib.dir/ylmr2.f90.o CMakeFiles/qe_upflib.dir/dylmr2.f90.o CMakeFiles/qe_upflib.dir/init_us_2_base_gpu.f90.o CMakeFiles/qe_upflib.dir/interp_atwfc_gpu.f90.o CMakeFiles/qe_upflib.dir/qvan2_gpu.f90.o CMakeFiles/qe_upflib.dir/sph_bes_gpu.f90.o CMakeFiles/qe_upflib.dir/ylmr2_gpu.
             f90.o CMakeFiles/qe_upflib.dir/dylmr2_gpu.f90.o CMakeFiles/qe_upflib.dir/simpsn_gpu.f90.o

     ...

     1923    cd /mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4/external && /anfhome/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placehold/linux-almalinux8-x86_64_v3/gcc-12.2.0/cmake-3.26.3-6uh2u4
             ctjz3lzaooy6w3ammknh56xysq/bin/cmake -E cmake_link_script CMakeFiles/qe_wannier90.dir/link.txt --verbose=1
     1924    /bin/ar qc ../lib/libqe_wannier90.a CMakeFiles/qe_wannier90.dir/wannier90/src/comms.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/constants.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/disentangle.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/hamiltonian.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/io.F90.o CMakeFiles/qe_
             wannier90.dir/wannier90/src/kmesh.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/overlap.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/parameters.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/plot.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/berry.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/boltzwann.F90
             .o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/dos.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/geninterp.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/get_oper.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/gyrotropic.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/kpath.F90.o CMakeFiles
             /qe_wannier90.dir/wannier90/src/postw90/kslice.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/postw90_common.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/spin.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/postw90/wan_ham.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/sitesym.F90.o CMakeFiles/qe_wannier90.dir/
             wannier90/src/transport.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/utility.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/wannierise.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/wannier_lib.F90.o CMakeFiles/qe_wannier90.dir/wannier90/src/ws_distance.F90.o
     1925    /bin/ranlib ../lib/libqe_wannier90.a
     1926    make[2]: Leaving directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
     1927    [ 14%] Built target qe_wannier90
     1928    make[1]: Leaving directory '/mnt/nvme/tmpscratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-4ktzyk4'
  >> 1929    make: *** [Makefile:149: all] Error 2

See build log for details:
  /mnt/scratch/root/spack-stage/spack-stage-q-e-sirius-develop-ristretto-4ktzyk4gdmk6hbsksnbkqdjpcwewszxj/spack-build-out.txt
</pre></details>

### Information on your system

* **Spack:** 0.21.0.dev0 (722b26094857337a045ef35b58473dae6e8a7621)
* **Python:** 3.6.8
* **Platform:** linux-almalinux8-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/12487790/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/12487791/spack-build-env.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by simonpintarelli** [comment]
This should be fixed in https://github.com/spack/spack/pull/39744.

**Comment by lorisercole** [comment]
Fixed by #39744
